cdap-watchdog/src/test/java/io/cdap/cdap/metrics/collect/AggregatedMetricsCollectionServiceTest.java

```diff
@@ -194,4 +199,80 @@ private void verifyGaugeMetricsValue(BlockingQueue<MetricValues> published, Map<
     // validate that no further aggregates are in the queue
     Assert.assertNull(published.poll());
   }
+
+  @Category(SlowTests.class)
+  @Test
+  public void testPublishDistribution() throws InterruptedException {
+    final BlockingQueue<MetricValues> published = new LinkedBlockingQueue<>();
+
+    AggregatedMetricsCollectionService service = new AggregatedMetricsCollectionService(1000L) {
+      @Override
+      protected void publish(Iterator<MetricValues> metrics) {
+        Iterators.addAll(published, metrics);
+      }
+    };
+
+    service.startAndWait();
+
+    // non-empty tags.
+    final Map<String, String> baseTags = ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, NAMESPACE,
+            Constants.Metrics.Tag.APP, APP,
+            Constants.Metrics.Tag.SERVICE, SERVICE,
+            Constants.Metrics.Tag.RUN_ID, RUNID);
+
+    try {
+      // publish distribution with empty tags
+      service.getContext(EMPTY_TAGS).event(DISTRIBUTION_METRIC, 40);
+      service.getContext(EMPTY_TAGS).event(DISTRIBUTION_METRIC, 9);
+      service.getContext(EMPTY_TAGS).event(DISTRIBUTION_METRIC, 57);
+      service.getContext(EMPTY_TAGS).event(DISTRIBUTION_METRIC, 200000);
+
+      verifyDistribtionMetricValues(published, 1, 4);
+
+      MetricsContext baseCollector = service.getContext(baseTags);
+      MetricsContext metricsContext = baseCollector.childContext(Constants.Metrics.Tag.HANDLER, HANDLER)
+              .childContext(Constants.Metrics.Tag.INSTANCE_ID, INSTANCE);
+
+      //publish distribution with non empty tags
+      baseCollector.event(DISTRIBUTION_METRIC, 40);
+      metricsContext.event(DISTRIBUTION_METRIC, 40);
+      baseCollector.event(DISTRIBUTION_METRIC, 9);
+      metricsContext.event(DISTRIBUTION_METRIC, 9);
+      baseCollector.event(DISTRIBUTION_METRIC, 57);
+      metricsContext.event(DISTRIBUTION_METRIC, 57);
+      baseCollector.event(DISTRIBUTION_METRIC, 200000);
+      metricsContext.event(DISTRIBUTION_METRIC, 200000);
+
+      verifyDistribtionMetricValues(published, 2, 4);
+
+    } finally {
+      service.stopAndWait();
+    }
+  }
+
+  private void verifyDistribtionMetricValues(BlockingQueue<MetricValues> published,
+                                             int numMetrics, int numValuesInMetric) throws InterruptedException {
+    long timeout = System.currentTimeMillis() + TimeUnit.SECONDS.toMillis(10);
+    MetricValues metricValues;
+    HashSet<Integer> tagCounts = new HashSet<>();
+    while (timeout > System.currentTimeMillis() && tagCounts.size() != numMetrics) {
+      metricValues = published.poll(100, TimeUnit.MILLISECONDS);
+      if (metricValues == null) {
+        continue;
+      }
+
+      Collection<MetricValue> values = metricValues.getMetrics();
+      for (MetricValue mv:values) {
```

Comment:
Please validate other MetricValue fields as well

Reply:
DistributionTest actually tests each property of Distribution. In going back and forth between mean and sum, I left it as mean accidentally there as well. Fixed it. This test is more about ensuring the metrics are getting aggregated with the right context.